src/sdk/namespace_fs.js

```diff
@@ -711,14 +724,13 @@ class NamespaceFS {
         // IMPORTANT NOTICE - we must return an etag that contains a dash!
         // because this is the criteria of S3 SDK to decide if etag represents md5
         // and perform md5 validation of the data.
-        const ident_str = 'inode-' + stat.ino.toString() + '-mtime-' + stat.mtimeMs.toString();
+        const ident_str = 'inode-' + stat.ino.toString() + '-mtime-' + stat.mtime.toString();
```

Comment:
what is the resolution of the mtime we return from stat?

Reply:
The timestamp is in seconds

Reply:
#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
	struct  timespec st_atimespec;  /* time of last access */
	struct  timespec st_mtimespec;  /* time of last data modification */
	struct  timespec st_ctimespec;  /* time of last status change */
#else
	time_t          st_atime;       /* [XSI] Time of last access */
	long            st_atimensec;   /* nsec of last access */
	time_t          st_mtime;       /* [XSI] Last data modification time */
	long            st_mtimensec;   /* last data modification nsec */
	time_t          st_ctime;       /* [XSI] Time of last status change */
	long            st_ctimensec;   /* nsec of last status change */
#endif

As you can see I can take the value from `st_mtimensec` but not on darwin, so I will do an if there.

Reply:
`
#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
	struct  timespec st_atimespec;  /* time of last access */
	struct  timespec st_mtimespec;  /* time of last data modification */
	struct  timespec st_ctimespec;  /* time of last status change */
#else
	time_t          st_atime;       /* [XSI] Time of last access */
	long            st_atimensec;   /* nsec of last access */
	time_t          st_mtime;       /* [XSI] Last data modification time */
	long            st_mtimensec;   /* last data modification nsec */
	time_t          st_ctime;       /* [XSI] Time of last status change */
	long            st_ctimensec;   /* nsec of last status change */
#endif
`
And this is the alias
`
#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
#define st_atime st_atimespec.tv_sec
#define st_mtime st_mtimespec.tv_sec
#define st_ctime st_ctimespec.tv_sec
#define st_birthtime st_birthtimespec.tv_sec
#endif /* (!_POSIX_C_SOURCE || _DARWIN_C_SOURCE) */
`
As you can see on darwin we do not have the option for nanoseconds, but I will add a code that will pull from nanoseconds as well and on darwin, I will just multiply for our local tests to pass.

Reply:
Darwin uses `struct timespec` which provides nano seconds too:
```
$ tail /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types/_timespec.h
#define _STRUCT_TIMESPEC        struct timespec

#include <machine/types.h> /* __darwin_time_t */

_STRUCT_TIMESPEC
{
	__darwin_time_t tv_sec;
	long            tv_nsec;
};
#endif /* _STRUCT_TIMESPEC */
```

Reply:
In any case - returning seconds is wrong - we need to convert to ms in order to be compatible with js dates.